frida_mode/src/cmplog/cmplog.c

```diff
@@ -81,6 +130,27 @@ gboolean cmplog_is_readable(guint64 addr, size_t size) {
 
   }
 
+  /*
+   * Our address map can change (e.g. stack growth), use write as a fallback to
+   * validate our address.
+   */
+  ssize_t written = syscall(__NR_write, fd_null, (void *)addr, size);
+
+  /*
+   * Seek back to the start, we don't want our temp file to continue growing!
+   */
+  if (ftruncate(fd_null, 0) < 0) {
```

Comment:
you want to truncate for every write?? rather do that once per fork() or persistent run

Reply:
I went with tracking the file size and truncating when it reached a given limit. It kept the interface between complog and the rest of the code more minimal and should still have the desired effect.